manifests/docker_run.pp

```diff
@@ -40,18 +49,16 @@
   }
 
   $image_tag = "${image}:${imagetag}"
-  docker::image { "${name}_${image_tag}" :  # make it possible to use the same docker image more than once on a node
-    image   => $image_tag,
-    require => Class['sunet::dockerhost'],
+  if $docker_image {
```

Comment:
The name $docker_image is misleading, but rather than changing that why don't you use ensure_resource('Docker::Image', ...) instead?

Reply:
Whats the difference between the current construct and running ensure_resource?

We can't run docker::image for localhost:5000/sunet/docker-registry-auth , when the registry container isn't up yet, and that happens towards the end of the puppet run.

run_docker_image ? fetc_docker_image ? or some other better idea?

Reply:
ensure_resource can avoid conflicting/multiple resource declarations, but that's not the issue here. We just need to turn off the call to docker::image.

Reply:
ok, then don't call the argument $docker_image since that name is commonly used as the name of the docker image

Reply:
Ok, renamed it to fetch_docker_image.